databricks_cli/jobs/cli.py

```diff
@@ -0,0 +1,131 @@
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from json import loads as json_loads
+
+import click
+from tabulate import tabulate
+
+from databricks_cli.click_types import OutputClickType
+from databricks_cli.jobs.api import create_job, list_jobs, delete_job, get_job, reset_job, run_now
+from databricks_cli.utils import eat_exceptions, CONTEXT_SETTINGS, pretty_format
+from databricks_cli.configure.config import require_config
+from databricks_cli.version import print_version_callback
+
+
+def _create_reset_base(json_file, json, api):
+    if not bool(json_file) ^ bool(json):
+        raise RuntimeError('Either --json-file or --json should be provided')
+    if json_file:
+        with open(json_file, 'r') as f:
+            json = f.read()
+    res = api(json_loads(json))
+    click.echo(pretty_format(res))
+
+
+@click.command(context_settings=CONTEXT_SETTINGS)
+@click.option('--json-file', default=None,
+              help='File containing json to POST to /jobs/create.')
+@click.option('--json', default=None, help='Displays absolute paths.')
+@require_config
+@eat_exceptions
+def create_cli(json_file, json):
+    _create_reset_base(json_file, json, create_job)
+
+
+@click.command(context_settings=CONTEXT_SETTINGS)
+@click.option('--json-file', default=None,
+              help='File containing json to POST to /jobs/create.')
+@click.option('--json', default=None, help='Displays absolute paths.')
+@require_config
+@eat_exceptions
+def reset_cli(json_file, json):
+    _create_reset_base(json_file, json, reset_job)
+
+
+def _jobs_to_table(jobs_json):
+    ret = []
+    for j in jobs_json['jobs']:
+        ret.append((j['job_id'], j['settings']['name']))
+    return ret
+
+
+@click.command(context_settings=CONTEXT_SETTINGS)
+@click.option('--output', default=None, help=OutputClickType.help, type=OutputClickType())
+@require_config
+@eat_exceptions
+def list_cli(output):
+    jobs_json = list_jobs()
+    if OutputClickType.is_json(output):
+        click.echo(pretty_format(jobs_json))
+    else:
+        click.echo(tabulate(_jobs_to_table(jobs_json), tablefmt='plain'))
+
+
+@click.command(context_settings=CONTEXT_SETTINGS)
+@click.option('--job-id', required=True, type=int)
+@require_config
+@eat_exceptions
+def delete_cli(job_id):
+    delete_job(job_id)
+
+
+@click.command(context_settings=CONTEXT_SETTINGS)
+@click.option('--job-id', required=True, type=int)
+@require_config
+@eat_exceptions
+def get_cli(job_id):
+    click.echo(pretty_format(get_job(job_id)))
+
+
+@click.command(context_settings=CONTEXT_SETTINGS)
+@click.option('--job-id', required=True, type=int)
+@click.option('--jar-params', default=None)
+@click.option('--notebook-params', default=None)
```

Comment:
how about python jobs and spark submit job?

Reply:
Let's add them once we document these fields in the API.